src/cargo/ops/cargo_update.rs

```diff
@@ -271,7 +276,14 @@ fn upgrade_dependency(
         return Ok(dependency);
     }
 
-    if !to_update.is_empty() && !to_update.contains(&name.to_string()) {
+    if !to_update.is_empty()
+        && !to_update.iter().any(|spec| {
+            spec.name() == name.as_str()
+                && spec
```

Comment:
We should check the spec has a registry source and if its for the  same registry.

Reply:
Good point. Fixed.